### ISSUE TITLE
release 1.4.0?

### ISSUE THREAD

**Issue by aph42** [pygeode/pygeode#135]
I'd like to put out a new release that includes the cartopy functionality just merged into the master branch, as well as some of the netcdf4 fixes (in particular the packing/unpacking issue). This is so that we can use cartopy for the upcoming pygeode short course.

@neishm, what would you think of merging the master branch into the release-1.3 branch as the basis for 1.3.2?

**Comment by neishm** [comment]
If there's new functionality, then we should probably have a 1.4.0 release to follow the semantic versioning guidelines.

**Comment by aph42** [comment]
Ok, that makes sense. I'll create a new release-1.4 branch in that case.

**Comment by aph42** [comment]
I've now gone ahead and created a release-1.4 branch. I am going to start using it for developing the short course, and will see how things go. I know this is a quick turn around but I am hoping we can push this out by Monday or Tuesday so that we can send out setup instructions.

**Comment by neishm** [comment]
Sounds good.  If you need some binary wheels uploaded to PyPI for JASMIN testing I could do that probably late Monday or Tuesday evening.

On a tangent, should "netcdf4" be added as a dependency for PyGeode, so it's available out of the box? Some environments like JASMIN aren't compatible with our old pygeode.formats.netcdf, since libnetcdf isn't installed on the host OS.

**Comment by aph42** [comment]
Thanks, that would be great. With regards to netcdf4 - I think that makes sense, but if we are going to do that maybe this is the right time to switch over to netcdf4 as the default format for netcdf files as well?

**Comment by neishm** [comment]
Sure, as long as the behaviour of open/close is unchanged from the end user's perspective that would be a good idea.  I've updated #117 to target release 1.4 now.

**Comment by aph42** [comment]
I think this should be the case, but given the fact that packing did not work as expected it's always possible that there are still different behaviours. I've made the switch in `formats/__init__.py`. I think this is potentially ready to go - I am going to try updating the conda recipe tonight. I'll update the docs as well. Can you try to upload this new version to PyPI?

**Comment by neishm** [comment]
I've uploaded an alpha release to PyPI, should be available by running
```
pip install pygeode --upgrade --pre
```
There was one regression test failure when switching to netcdf4, which should now be resolved.

I'm using a pre-release version for the PyPI upload since we'll only have one shot at uploading a final 1.4.0 release (we can't overwrite the existing files if further changes are needed).

**Comment by aph42** [comment]
I've been working with this all day in preparation for the short course; it
does everything we need it to so I think we're good to do the production
run, from my perspective.

**Comment by neishm** [comment]
Okay, final version uploaded.

**Comment by aph42** [comment]
I've pushed docs that are consistent with this release to github.io. I (of course) found one stupid issue that I introduced accidentally in the plotting module in how some aspects of the contouring are chosen automatically. I've pushed a commit that fixes the bug. It's not critical (we can live with it for the short course), but it's there. I did so before thinking of adding a tag 1.4.0 to the previous commit which we should probably do, then maybe we can close this issue?
